### PR TITLE
Server side implementation for multiple rooms.

### PR DESCRIPTION
**Pull request by HitRam**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -17,24 +17,32 @@ app.use('/',index);
...
 		else if(users.indexOf(name) == -1) {
-			users.push(name.trim());
+			users.push(name);
 
 			//username is valid so user is set
 			socket.emit('user set', name);
 
+			//by default, user joins lobby
+			socket.join('lobby');
+			rooms[0][0]++;
```
Comment (abhigyank) [inline app.js]: should be rooms[0][1].

File: app.js [hunk 2]
```diff
@@ -17,24 +17,32 @@ app.use('/',index);
...
+
 			//notify all users (except sender) that user joined
-			socket.broadcast.emit('user joined', name);
+			socket.broadcast.emit('user joined', {username: name, num_users: rooms[0][0]});
```
Comment (abhigyank) [inline app.js]: again rooms[0][1].

**Review by abhigyank — CHANGES_REQUESTED** [review]

File: app.js [hunk 3]
```diff
@@ -47,16 +55,100 @@ io.on('connection', function(socket) {
...
+	});
+
 	//When user disconnets remove user from users
 	socket.on('disconnect', function() {
 
+		var num_rooms = rooms.length;
+
+		//update number of users in rooms 
+		for(var i = 0; i < num_rooms; i++) {
+			if(socket.rooms.indexOf(rooms[i][0]) >= 0) {
```
Comment (abhigyank) [inline app.js]: Have you tested this?
Once disconnected I think socket.rooms would return an empty array.

Comment (HitRam) [reply app.js]: I didn't test this. Wait I'll check.

Comment (HitRam) [reply app.js]: It works now.

**Pull request merged**